documentation/specs/static-graph.md

```diff
@@ -237,16 +310,43 @@ There is also the possibility for these higher-order build engines and even Visu
 
-## Distribution
-To support distribution for isolated builds (eg. for the MS internal build engine), we need a solution for a project and a dependency building on different machines. To facilitate this, we need to serialize the build result cache to disk on the source machine which built the project reference, and deserialize it on the destination machine which is about to build the project referencing the project reference.
+### Single project isolated builds: implementation details
+
+<!-- workflow -->
+Single project builds can be achieved by providing MSBuild with input and output cache files.
+
+The input cache files contain the cached results of all of the current project's references. This way, when the current project executes, it will naturally build its references via [MSBuild task](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-task) calls. The engine, instead of executing these tasks, will serve them from the provided input caches. 
+
+The output cache file tells MSBuild where it should serialize the results of the current project. This output cache would become an input cache for all other projects that depend on the current project.
+The output cache file can be ommited in which case the build would just reuse prior results but not write out any new results. This could be useful when one wants to replay a build from previous caches.
+
+The presence of either input or output caches turns on the isolated build constraints. The engine will fail the build on:
+- MSBuild task calls on projects files which were not defined in the `ProjectReference` item at evaluation time.
+- MSBuild task calls which cannot be served from the cache
+
+<!-- cache structure -->
+These cache files contain the serialized state of MSBuild's [ConfigCache](https://github.com/Microsoft/msbuild/blob/master/src/Build/BackEnd/Components/Caching/ConfigCache.cs) and [ResultsCache](https://github.com/Microsoft/msbuild/blob/master/src/Build/BackEnd/Components/Caching/ResultsCache.cs). These two caches have been traditionally used by the engine to cache build results.
+
+Cache structure: `(project path, global properties) -> results`
+
+<!-- cache lifetime -->
+The caches are applicable for the entire duration of the MSBuild.exe process. The input and output caches have the same lifetime as the `ConfigCache` and the `ResultsCache`. The `ConfigCache` and the `ResultsCache` are owned by the [BuildManager](https://github.com/Microsoft/msbuild/blob/master/src/Build/BackEnd/BuildManager/BuildManager.cs), and their lifetimes are one `BuildManager.BeginBuild` / `BuildManager.EndBuild` session. Since MSBuild.exe uses one BuildManager with one BeginBuild / EndBuild session, the cache lifetime is the same as the entire process lifetime.
+
+<!-- constraints -->
+The following are some constraints enforced by the engine. They're just for 
```

Comment:
Just for?

Reply:
Just for showing my sometimes short attention span :)